app/models/resource_type.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class ResourceType < ActiveRecord::Base
+  validates :name, presence: true, uniqueness: true
+  validates :dtd_file, presence: true
+
+  after_validation :dtd_file_exists
```

Comment:
Since it is doing validation, I wanted to use one of the `x_validation` callbacks as opposed to `x_save`.  I want it to be done on creates and updates.